--- a/lib/socket.c
+++ b/lib/socket.c
@@ -74,6 +74,22 @@
 #include <assert.h>
 
 #include "ucode/module.h"
+
+#ifndef IPV6_MULTICAST_ALL
+#define IPV6_MULTICAST_ALL 29
+#endif
+
+#ifndef IPV6_ROUTER_ALERT_ISOLATE
+#define IPV6_ROUTER_ALERT_ISOLATE 30
+#endif
+
+#ifndef TCP_CM_INQ
+#define TCP_CM_INQ 36
+#endif
+
+#ifndef UDP_GRO
+#define UDP_GRO 104
+#endif
 #include "ucode/platform.h"
 
 #if defined(__linux__)
